### PR TITLE
Depend on our own custom checker framework fork for now.

### PR DESCRIPTION
**Pull request by lazaroclapp**
This commit:

- Adds https://github.com/msridhar/checker-framework
  (repository branch) as a maven repository for our build.gradle
- Forces the 2.1.14-nullaway checker framework dataflow jar
  (but uses upstream javacutil.jar for same version). Forcing
  needed, as 2.1.14-* < 2.1.14 for dependency resolution.
- Adds the shadow maven plug-in allowing us to create a semi-fat
  jar using `gradle shadowJar` which contains NullAway and the
  checker framework fork, but not, say, EP or javax.
- Removes our hack around a CF fix we backported into our fork:
  https://github.com/typetools/checker-framework/pull/1516

### CODE REVIEW

File: nullaway/build.gradle [hunk 1]
```diff
@@ -26,11 +27,20 @@ configurations.maybeCreate("epJavac")
...
+        force deps.build.checkerDataflow
+    }
+}
+
 dependencies {
     compileOnly deps.apt.autoValue
     compileOnly deps.apt.autoService
 
-    compile deps.build.errorProneCheckApi
+    shadow deps.build.errorProneCheckApi
```
Comment (lazaroclapp) [inline nullaway/build.gradle]: For some reason, saying `shadow` here means "exclude this and its dependencies from the shadowJar"... yes, really...

File: nullaway/build.gradle [hunk 2]
```diff
@@ -47,6 +57,10 @@ dependencies {
     epJavac deps.build.checkerDataflow
 }
 
+shadowJar {
+    classifier = "cf"
```
Comment (lazaroclapp) [inline nullaway/build.gradle]: Our NullAway+CF jar will be called `nullaway-version-cf.jar`

For now, I am excluding most dependencies, only bundling CF-dataflow and CF-javacutil. But let me know if we want to make a true fat jar, with EP and javax and everything.

File: nullaway/src/main/java/com/uber/nullaway/dataflow/DataFlow.java [hunk 3]
```diff
@@ -94,16 +94,14 @@ public ControlFlowGraph load(CfgParams key) {
...
+                                        bodyPath = new TreePath(methodPath, lambdaExpressionTree.getBody());
                                     } else {
                                         // must be a method per findEnclosingMethodOrLambda
                                         MethodTree method = (MethodTree) methodPath.getLeaf();
                                         ast = new UnderlyingAST.CFGMethod(method, /*classTree*/ null);
                                         bodyPath = new TreePath(methodPath, method.getBody());
-                                        return CFGBuilder.build(
-                                                bodyPath, env, ast, false, false);
                                     }
+                                    return CFGBuilder.build(bodyPath, env, ast, false, false);
```
Comment (lazaroclapp) [inline nullaway/src/main/java/com/uber/nullaway/dataflow/DataFlow.java]: 😄

File: build.gradle [hunk 4]
```diff
@@ -25,6 +25,9 @@ subprojects {
     }
 
     repositories {
+        maven {
+            url "https://raw.githubusercontent.com/msridhar/checker-framework/repository/"
```
Comment (lazaroclapp) [inline build.gradle]: Tested on VPN, no problems so far.

**Comment by lazaroclapp** [comment]
It seems `:sample` is not seeing the version of `:nullaway` with the shadowed CF classes. Simply making `shadowJar.classifier = ""` and having `build.dependsOn shadowJar` does not solve this issue, so I'll dig into it a bit more tomorrow. Unless either of you know what exactly `apt project(":nullaway")` pulls in.

File: nullaway/build.gradle (hunk 1 above)
Comment (msridhar) [reply nullaway/build.gradle]: weird...

File: nullaway/build.gradle (hunk 2 above)
Comment (msridhar) [reply nullaway/build.gradle]: No, this is right, we just want to bundle the CF classes

Comment (kageiit) [reply nullaway/build.gradle]: is the classifier here a maven classifier?

**Comment by msridhar** [comment]
Re: the build failure, not sure what is going on.  We need to say somehow that the `sample` project depends on the shadow jar output of the `nullaway` project; not sure how to do that.

**Comment by kageiit** [comment]
> Re: the build failure, not sure what is going on. We need to say somehow that the sample project depends on the shadow jar output of the nullaway project; not sure how to do that.

You can say

```
compile project(name: "nullaway", configuration: "shadow")
```

**Comment by lazaroclapp** [comment]
This alone does not solve the problem:

```
apt project(path: ":nullaway", configuration: "shadow")
```

When we depend on another project, how do the clases of that project get loaded? Is it just that the project gets built and then the `nullaway-0.1.5-SNAPSHOT.jar` added to the classpath? Because if so, we need to substitute that with `nullaway-0.1.5-SNAPSHOT-cf.jar`. 

However, I tried setting `shadowJar.classifier = null`, which removes the `-cf` part and makes `nullaway-0.1.5-SNAPSHOT.jar` our fat jar, but this does still not get us the right classes at runtime.

File: nullaway/build.gradle (hunk 2 above)
Comment (lazaroclapp) [reply nullaway/build.gradle]: I am not sure, it gets added at the end of that jar filename, I am not sure were else it is used. I'd be fine with setting it to `null` and having both `shadow` be our default build and this being our default jar.

**Pull request merged**